### ISSUE TITLE
Segmentation fault after building with SWIG 4.0

### ISSUE THREAD

**Issue by VladYagl** [automl/random_forest_run#51]
New version of SWIG (https://github.com/swig/swig/releases/tag/rel-4.0.0) breaks any function call from package:
`python terminated by signal SIGSEGV (Address boundary error) 139`

Works fine with SWIG 3.0.12.

**Comment by shukon** [comment]
@AndreBiedenkapp @mlindauer this creates problems for Ubuntu 20.04. Proposed solution worked after fiddling with my system forever... Had to uninstall swig, remove all links to it in `/usr/bin/swig` and `/usr/bin/swig4-0`, install `apt install swig3-0`, `ln -s /usr/bin/swig3.0 /usr/bin/swig`, recreate virtual environment, reinstall python (though I don't think that was necessary) and most importantly reinstall cave with the `--no-cache-dir` flag of pip... so this is not an arch-linux phenomenon anymore.

The actual fix is likely easier. I assume it's the proposed solution below, but I didn't test it because it's working on my system now and I'm not inclined to risk it...
```
sudo apt remove swig -y
sudo apt install swig3.0
sudo ln -s /usr/bin/swig3.0 /usr/bin/swig
# activate virtual environment
pip install pyrfr --no-cache-dir
```

Anyway, it's not some arch-linux cornercase anymore.

**Comment by mfeurer** [comment]
Another workaround is using Anaconda to separate from the system swig, as for example in the SMAC3 unit tests (https://github.com/automl/SMAC3)

**Comment by mlindauer** [comment]
I agree with @mfeurer that we should strongly recommend anaconda in such a case. I would not like to recommend to people how to fiddle with their systems, by uninstalling stuff only such that the pyrfr works again.

**Comment by shukon** [comment]
Fair enough. I just figured I'd leave this workaround documented here and mention you so you're informed, but I agree that anaconda is the better alternative - or, at some point, updating pyrfr to swig 4, since I believe that swig 3 is not getting security updates or anything anymore.
But, yes, until then anaconda works great.

**Comment by mfeurer** [comment]
Fixed by #61
